R/plot_star_system.R

```diff
@@ -48,13 +57,25 @@ plot_star_system = function(planet_data, spectral_type = NULL) {
   #    - ensure star is the largest object with size 17
   #    - map color based on spectral type or density
   plot_data = data.frame(
-    orbit_offset = c(0, runif(nrow(planet_data), min = 0, max = 2 * pi)),
-    pl_orbsmax = c(0, .rescale_orbsmax(planet_data$pl_orbsmax)),
-    pl_rade = c(17, planet_data$pl_rade), # Ensure star is the largest object
-    col = c(.map_star_color(spectral_type), .map_planet_color(planet_data$pl_dens))
+    orbit_offset = c(0, runif(nrow(planet_data), min = 0, max = 2 * pi), 0, 0),
+    pl_orbsmax = c(0, .rescale_orbsmax(c(planet_data$pl_orbsmax,
+                                         # Add habitable zone edges for scaling
+                                         min(habitable_zone), max(habitable_zone)))),
```

Comment:
style: comment should be in the comment block above
style: indentation
```suggestion
    pl_orbsmax = c(0, .rescale_orbsmax(c(
      planet_data$pl_orbsmax,
       min(habitable_zone), 
       max(habitable_zone)
     ))),
```